plane_mcp/resolver.py

```
@@ -0,0 +1,168 @@
+        if key in self._work_item_cache and (time.time() - self._ts["work_items"] < _CACHE_TTL_SECONDS):
+            return self._work_item_cache[key]
+        
+        parts = key.split('-')
+        if len(parts) != 2:
+            raise ValueError(f"Invalid ticket ID format: '{ticket_id}'. Expected format like 'ENG-123'.")
+        
+        project_identifier = parts[0]
+        try:
+            issue_sequence = int(parts[1])
+        except ValueError:
+            raise ValueError(f"Invalid ticket sequence in '{ticket_id}'. Must be an integer (e.g., 123).")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)